lib/extractor/layer.ts

```diff
@@ -135,6 +201,48 @@ function getLayersContentAndArchiveManifest(
   };
 }
 
+function getOciLayersContentAndArchiveManifest(
+  imageIndex: OciImageIndex,
+  manifestCollection: OciArchiveManifest[],
+  layers: Record<string, ExtractedLayers>,
+): { layers: ExtractedLayers[]; manifest: OciArchiveManifest } {
+  // filter empty layers
+  // get the layers content without the name
+  // reverse layers order from last to first
+
+  // get manifest file first
+  const manifest = getOciManifest(imageIndex, manifestCollection);
+  const filteredLayers = manifest.layers
+    .filter((layer) => Object.keys(layers[layer.digest]).length !== 0)
+    .map((layer) => layers[layer.digest])
+    .reverse();
+
+  return {
+    layers: filteredLayers,
+    manifest,
+  };
+}
+
+function getOciManifest(
+  imageIndex: OciImageIndex,
+  manifestCollection: OciArchiveManifest[],
+): OciArchiveManifest {
+  if (imageIndex.manifests.length === 1) {
+    return manifestCollection[0];
+  }
+
+  const manifest = imageIndex.manifests.find(
+    (item) =>
+      item.platform?.architecture === "amd64" && item.platform.os === "linux",
+  );
+
+  if (manifest === undefined) {
+    throw new Error("Unsupported type of CPU architecture or operating system");
```

Comment:
🤔 I will test it!

Reply:
I think we need to ensure all functions called in the `.on()` handlers are exception safe. If they do throw we can do `catch (error) { reject(error); }` so that we can properly return the exception to the caller of `await scanOciArchive()` (or whatever the name was 😛)

Reply:
So I actually have tried that before and result is exactly the same! I am getting an unhandled rejection, there's two ways you reject the promise right, one is by calling `reject()` and one when an error is thrown, so I think this is ok 🤔

Reply:
So that code above gives exactly the same results as: 

```
      try {
      resolve(
        getOciLayersContentAndArchiveManifest(imageIndex, manifests, layers),
      );
      } catch (error) {
        reject(error);
      }
```

Reply:
It is working as expected after properly testing :-)